pyqtgraph/functions.py

```diff
@@ -1752,51 +1728,104 @@ def arrayToQPath(x, y, connect='all'):
             if first < len(x):
                 # Replace all non-finite entries from beginning of arr with the first finite one
                 idx[:first] = first
-                arr[1:-1] = arr[1:-1][idx]
+                arr[:] = arr[:][idx]
 
     # decide which points are connected by lines
-    if eq(connect, 'all'):
-        arr[1:-1]['c'] = 1
+    if eq(connect, 'all') or (eq(connect, 'finite') and QT_LIB in ["PySide6", "PyQt6"]):
+        # turns out with Qt6 we can give the QPolygonF NaN values and it works out!
```

Comment:
If we enter here due to "finite" and Qt6, then the finite checking above needs to bypassed

Reply:
Decided this was too confusing; reverted to a separate `elif eq(connect, 'finite')` logic, with separate branches for Qt6 or not...